common/internal_model/src/test/scala/uk/ac/wellcome/models/parse/DateParserTest.scala

```diff
@@ -0,0 +1,102 @@
+package uk.ac.wellcome.models.parse
+
+import org.scalatest.{FunSpec, Matchers}
+
+import java.time.LocalDate
+
+import uk.ac.wellcome.models.work.internal.InstantRange
+import uk.ac.wellcome.models.parse.parsers.DateParser
+
+class DateParserTest extends FunSpec with Matchers {
+
+  it("parses year") {
```

Comment:
Would http://www.scalatest.org/user_guide/table_driven_property_checks help to make these tests more succinct?

Reply:
I did use that at first, but I ended up preferring this method for readability. Don't mind too much either way.

Reply:
Feels like consistency might be better here as a method of testing, unless it's really unreadable?

Reply:
I have had a think and actually think this is better than using the table based tests for these test cases.

If we use table based test the error messages will be quite hard to parse, whereas here the textual descriptions make it clear (e.g. seeing "parses written date ranges spanning multiple years" has failed rather than "`Some(InstantRange(LocalDateTime(...))` does not equal `Some(InstantRange(LocalDateTime(...))`").

Reply:
You could have one of the table columns being the description of the text you are trying to parse... not sure if that's wedging it in though.

Reply:
```scala
Table(
  ("date ranges spanning multiple years", InstantRange(LocalDateTime(...), LocalDateTime(...))),
  ....
) { (description, instant) => ... }
```

Assuming these tests are going to fail now too (they're crap tests)
https://github.com/wellcometrust/catalogue/blob/master/common/internal_model/src/test/scala/uk/ac/wellcome/models/work/internal/InstantRangeTest.scala

Reply:
Have tried out the `Table` syntax and discussed with @jamesgorrie. We have decided to keep the syntax as is, due to slightly better error reporting and readability.